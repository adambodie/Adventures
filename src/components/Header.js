import React from 'react'
import '../styles/header.scss'
import { Link } from 'gatsby'
import FontAwesome from 'react-fontawesome'

const Header = () => (
	<header>
		<div className='header-logo'>
			<img src={'https://adventures.bodiewebdesign.com/_photos/logo.jpg'} alt='Logo' className='img-fluid' />
			<h1 data-testid='header-title'>Adam&apos;s Adventures</h1>
			<Link to='/pages'>
				<FontAwesome className='icon' name='home'/>
			</Link>
			<Link to='/tags'>
				<FontAwesome className='icon' name='tags'/>
			</Link>
		</div>
	</header>
)

export default Header;
