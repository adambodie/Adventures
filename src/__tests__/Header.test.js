import React from "react"
import renderer from "react-test-renderer"
import Header from "../components/Header"
import { render } from "@testing-library/react"

describe("Header", () => {
	it("renders correctly", () => {
		const tree = renderer.create(<Header  />).toJSON()
		expect(tree).toMatchSnapshot()
	})
	it("displays the correct title", () => {
		const { getByTestId } = render(<Header />)
		expect(getByTestId("header-title")).toHaveTextContent("Adam's Adventures")
	})
})
