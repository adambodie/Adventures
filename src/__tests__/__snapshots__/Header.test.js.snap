// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<header>
  <div
    className="header-logo"
  >
    <img
      alt="Logo"
      className="img-fluid"
      src="../logo.jpg"
    />
    <h1
      data-testid="header-title"
    >
      Adam's Adventures
    </h1>
    <a
      href="/pages"
    >
      <span
        aria-hidden={true}
        className="fa fa-home icon"
      />
    </a>
    <a
      href="/tags"
    >
      <span
        aria-hidden={true}
        className="fa fa-tags icon"
      />
    </a>
  </div>
</header>
`;
